transaction/simple.go

```diff
@@ -0,0 +1,72 @@
+package transaction
+
+import (
+	"github.com/gluster/glusterd2/context"
+)
+
+// SimpleTxn is transaction with fixed stage, commit and store steps
+type SimpleTxn struct {
+	// Ctx is the transaction context
+	Ctx *context.Context
+	// Nodes are the nodes where the stage and commit functions are performed
+	Nodes []string
+	// LockKey is the key to be locked
+	LockKey string
+
+	// Stage function verifies if the node can perform an operation.
+	Stage StepFunc
+	// Commit performs the operation on the a node
+	Commit StepFunc
+	// Store stores the results of an operation. This will only be run on the leader
+	Store StepFunc
+	// Rollback rollsback any changes done by Commit
+	Rollback StepFunc
+}
+
+// NewSimpleTxn returns creates and returns a Txn using e Simple transaction template
+func NewSimpleTxn(c *context.Context, nodes []string, lockKey string, stage, commit, store, rollback StepFunc) (*Txn, error) {
+	simple := Txn{
+		Ctx:   c,
+		Nodes: nodes,
+		Steps: make([]*Step, 5), //A simple transaction has just 5 steps
+	}
+
+	lockstep, unlockstep, err := CreateLockSteps(lockKey)
+	if err != nil {
+		return nil, err
+	}
+
+	stagestep := &Step{
+		DoFunc:   stage,
+		UndoFunc: nil,
+		Nodes:    []string{All},
+	}
+	commitstep := &Step{
+		DoFunc:   commit,
+		UndoFunc: rollback,
+		Nodes:    []string{All},
+	}
+	storestep := &Step{
+		DoFunc:   store,
+		UndoFunc: nil,
```

Comment:
Doing nothing in Undo here is a problem given that store is post commit, ideally we should use rollback, isn't it?